src/Compilers/CSharp/Portable/Symbols/ReferenceManager.cs

```diff
@@ -424,19 +424,20 @@ private bool CreateAndSetSourceAssemblyFullBind(CSharpCompilation compilation)
 
                     for (int i = 1; i < bindingResult.Length; i++)
                     {
-                        if ((object)bindingResult[i].AssemblySymbol == null)
+                        ref BoundInputAssembly bound = ref bindingResult[i];
```

Comment:
The lack of member tracking in arrays was particularly problematic for this change. Then again this code is essentially paassing around large arrays of mutable `struct` values. It's the worst possible case for nullability. This doesn't make me want to push for array index tracking (that's really expensive) but it does make me think that if we were doing this again we would've chosen a different approach. Possibly wrapping this array in a simple `struct` wrapper that could've handled many of these assertions.